ts-client/src/dlmm/index.ts

```diff
@@ -2007,6 +1646,78 @@ export class DLMM {
     }).add(setComputeUnitLimitIx, createPositionIx, ...createBinArrayIxs);
   }
 
+  /**
+   * The function `getPosition` retrieves position information for a given public key and processes it
+   * using various data to return a `LbPosition` object.
+   * @param {PublicKey} positionPubKey - The `getPosition` function you provided is an asynchronous
+   * function that fetches position information based on a given public key. Here's a breakdown of the
+   * parameters used in the function:
+   * @returns The `getPosition` function returns a Promise that resolves to an object of type
+   * `LbPosition`. The object contains the following properties:
+   * - `publicKey`: The public key of the position account
+   * - `positionData`: Position Object
+   * - `version`: The version of the position (in this case, `Position.V2`)
+   */
+  public async getPosition(positionPubKey: PublicKey): Promise<LbPosition> {
+    const positionAccountInfo = await this.program.account.positionV2.fetch(positionPubKey);
+    if (!positionAccountInfo) {
+      throw new Error(`Position account ${positionPubKey.toBase58()} not found`);
+    }
+
+    const { lowerBinId, upperBinId, feeOwner } = positionAccountInfo;
+    const lowerBinArrayIndex = binIdToBinArrayIndex(new BN(lowerBinId));
+    const upperBinArrayIndex = binIdToBinArrayIndex(new BN(upperBinId));
+    const [lowerBinArrayPubKey] = deriveBinArray(
+      this.pubkey,
+      lowerBinArrayIndex,
+      this.program.programId
+    );
+    const [upperBinArrayPubKey] = deriveBinArray(
+      this.pubkey,
+      upperBinArrayIndex,
+      this.program.programId
+    );
+
+    const [clockAccInfo, lowerBinArrayAccInfo, upperBinArrayAccInfo] = await chunkedGetMultipleAccountInfos(
+      this.program.provider.connection,
+      [
+        SYSVAR_CLOCK_PUBKEY,
+        lowerBinArrayPubKey,
+        upperBinArrayPubKey,
+      ]
+    );
+
+    const onChainTimestamp = new BN(
+      clockAccInfo.data.readBigInt64LE(32).toString()
+    ).toNumber();
+
+    const lowerBinArray = this.program.coder.accounts.decode(
```

Comment:
Would have the case that position is initialized, but lowerBinArray is not initialized
So position would have zero liquidity

Reply:
pushed a fix where if no bin array it will return position data with zero liquidity